Task1/Node/fs-demo.js

```diff
@@ -0,0 +1,45 @@
+const ghash = require('./hash-generation.js');
+
+const fs = require("fs");
+const path = fs.readdirSync("./utils");
+//TODO check if utils is path or directory ?
+var stats = fs.statSync("./utils");
+console.log('Is utlis a file or directory ?');
+console.log('is file ? ' + stats.isFile());
+console.log('is directory ? ' + stats.isDirectory());
+
+//TODO
+//  read all file names in the directory
+// if you find a directory inside a path you go read recursively
+
+console.log('All files and Directory names listed : ');
+async function print(path) {
+    const dir = await fs.promises.opendir(path);
+    for await (const dirent of dir) {
+
+        if (dirent.isDirectory()) {
+            //recursively calling to print file names inside directory
+            console.log(' directory  ' + dirent.name);
+            print(path + '/' + dirent.name.toString()).catch(console.error);
```

Comment:
for path concatenation its a good habit to use `path` module. It takes care of all Operating Systems.

Reply:
modified the code by making use of  path.join() function .